RingPublishingGDPR/src/main/java/com/ringpublishing/gdpr/RingPublishingGDPR.java

```diff
@@ -284,42 +318,90 @@ private boolean isOutdated()
         return outdated;
     }
 
-    private void verifyConsents(@NonNull Context applicationContext)
+    private void verifyOutdated()
     {
         final Map<String, String> consents = storage.getRingConsents();
 
         if (consents == null || consents.isEmpty())
         {
             Log.w(TAG, "Fail verify consents. Consents are empty");
+            setVerifyState(VerifyState.FAILURE);
             return;
         }
 
-        api.verify(consents, new VerifyApiCallback()
+        api.verify(consents, new VerifyCallback()
         {
             @Override
             public void onOutdated(String rawStatus)
             {
-                storage.saveLastAPIConsentsCheckStatus(rawStatus);
                 storage.setOutdated(true);
-                FormView formView = getFormView(applicationContext);
-                showFromApplication(formView);
+                storage.saveLastAPIConsentsCheckStatus(rawStatus);
+                setVerifyState(VerifyState.OUTDATED);
             }
 
             @Override
             public void onActual(String rawStatus)
             {
-                storage.saveLastAPIConsentsCheckStatus(rawStatus);
                 storage.setOutdated(false);
+                storage.saveLastAPIConsentsCheckStatus(rawStatus);
+                setVerifyState(VerifyState.ACTUAL);
             }
 
             @Override
-            public void onFail(String status)
+            public void onFailure(String status)
             {
                 storage.saveLastAPIConsentsCheckStatus(status);
+                setVerifyState(VerifyState.FAILURE);
             }
         });
     }
 
+    private void setVerifyState(VerifyState verifyState)
+    {
+        this.verifyState = verifyState;
+        apiCallFinished();
+    }
+
+    private synchronized void apiCallFinished()
```

Comment:
nie za bardzo ta klasa - co by nie mówić publiczna - rośnie nam tu z uwagi na prywatne metody?

Reply:
Teś mi trochę tu śmierdziało to ale używa dużo parametrów z klasy.

Jak nie będzie dużo trzeba wycinać to deleguję to do innej klasy.